pkgs/by-name/ne/newlisp/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  fetchurl,
+  stdenv,
+  libffi,
+  readline,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "newlisp";
+  version = "10.7.5";
+
+  src = fetchurl {
```

Comment:
`fetchzip` is better than `fetchurl`, see https://discourse.nixos.org/t/fetchzip-or-fetchurl-which-should-i-use/36572/2

This is just a suggestion which does not block the merge of this PR.

Reply:
But I guess that leaving the unpacking process to an explicit, (easier-to) injectable phase may provides a better interface.

Reply:
> leaving the unpacking process to an explicit, (easier-to) injectable phase may provides a better interface.

Could you give me a concrete example showing the benefits?

FYI, it indeed happened that upstream changed mtime of files in the tarball and thus invalidated our hash of `fetchurl`.

As I said above, I do not have a strong opinion on this and can merge this PR as is.

Reply:
No. Just a matter of sense. Have no strong preference, too.